--- a/ecell/frontend/model-editor/Makefile.am
+++ b/ecell/frontend/model-editor/Makefile.am
@@ -99,4 +99,4 @@
 	rm -rf build
 
 install-exec-local:
-	$(PYTHON_) setup.py install -f --prefix=$(PREFIX) --install-scripts=$(bindir) $(if $(DESTDIR),--root $(DESTDIR), )
+	$(PYTHON_) setup.py install -f --prefix=$(PREFIX) --install-scripts=$(bindir) --install-layout=deb --no-compile -O0 $(if $(DESTDIR),--root $(DESTDIR), )
--- a/ecell/frontend/session-monitor/Makefile.am
+++ b/ecell/frontend/session-monitor/Makefile.am
@@ -54,4 +54,4 @@
 	rm -rf build
 
 install-exec-local:
-	$(PYTHON_) setup.py install -f --prefix=$(PREFIX) --install-scripts=$(bindir) $(if $(DESTDIR),--root $(DESTDIR), )
+	$(PYTHON_) setup.py install -f --prefix=$(PREFIX) --install-scripts=$(bindir) --install-layout=deb --no-compile -O0 $(if $(DESTDIR),--root $(DESTDIR), )
--- a/ecell/pyecell/Makefile.am
+++ b/ecell/pyecell/Makefile.am
@@ -90,4 +90,4 @@
 	$(RM) -rf build
 
 install-exec-local:
-	$(PYTHON_) setup.py install -f --prefix=$(PREFIX) --install-scripts=$(bindir) $(if $(DESTDIR),--root $(DESTDIR), )
+	$(PYTHON_) setup.py install -f --prefix=$(PREFIX) --install-scripts=$(bindir) --install-layout=deb --no-compile -O0 $(if $(DESTDIR),--root $(DESTDIR), )
