From 8d8ddd7848744e3dea95d48c5d44b1a5335b7f57 Mon Sep 17 00:00:00 2001
From: Moriyoshi Koizumi <mozo@mozo.jp>
Date: Mon, 28 May 2012 18:17:58 +0900
Subject: [PATCH] Oops.

---
 dmtool/dmcompile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dmtool/dmcompile.in b/dmtool/dmcompile.in
index 81af612..3b0b0af 100644
--- a/dmtool/dmcompile.in
+++ b/dmtool/dmcompile.in
@@ -100,7 +100,7 @@ class CustomizedOptionParser( OptionParser ):
 
     def _process_long_opt(self, rargs, values):
         arg = rargs[ 0 ]
-        argname, _ = arg.split( '=', 1 )
+        argname, _, _ = arg.partition( '=' )
         try:
             OptionParser._process_long_opt( self, rargs, values )
         except BadOptionError:
-- 
1.7.10

